### PR TITLE
Fix for layernorm skip connections in sq

### PR DESCRIPTION
**Pull request by maktukmak**
## Type of Change

Bug fix

## Description

When folding is enabled in Smoothquant, layer norms are used as absorbing layers for linear layers. However, when there are skip connections from the layer_norms, i.e., the output of layer norm goes to multiple ops, smoothquant quant transform produces wrong results. 

An example graph is shown below:

image

which occurs in the BridgeTower model. This PR checks if any layer norm ops have multiple children, in which case, it is not used as absorbing layer.

## Expected Behavior & Potential Risk

Smoothquant transform won't produce incorrect results when an absorb layer norm has skip connections

## Dependency Change?

This PR works best after applying the following PRs: #961 and #954

### CODE REVIEW

**Review by xin3he — APPROVED** [review]

**Comment by chensuyue** [comment]
Please fix DCO and Pylint scan.

**Review by wenhuach21 — APPROVED** [review]

File: neural_compressor/adaptor/torch_utils/smooth_quant.py [hunk 1]
```diff
@@ -855,7 +855,14 @@ def get_prev_absorb_layer(self, nodes):
                     parent = get_parent(parent)
                     continue
                 if parent.kind() in self.could_absorb_layers:
-                    prev_absorb_layer.append(parent)
+                    # Check if parent has multiple children of unsupported layers
+                    set_inp_type = set(dict_inp[parent])
```
Comment (wenhuach21) [inline neural_compressor/adaptor/torch_utils/smooth_quant.py:859]: the code has some issues. We should check output nodes of  the parent

**Review by wenhuach21 — CHANGES_REQUESTED** [review]

File: neural_compressor/adaptor/torch_utils/smooth_quant.py [hunk 2]
```diff
@@ -875,10 +882,17 @@ def get_absorb_to_layer(self, model, example_input, op_types):
         traced_model = self.trace(model, example_input)
         if traced_model == None:
             return None, None
+        
+        dict_inp = defaultdict(list)
+        for node in traced_model.graph.nodes():
+            inp_list = list(node.inputs())
```
Comment (wenhuach21) [inline neural_compressor/adaptor/torch_utils/smooth_quant.py:888]: we should check the node's output

**Review by wenhuach21 — CHANGES_REQUESTED** [review]

**Review by yintong-lu — COMMENTED** [review]
Hi, I think there's a mistake when you are creating dict_inp. The input-types of all nodes are saved rather than the output-type.

**Comment by chensuyue** [comment]
Hi @maktukmak, since there are some issues in this PR, we submit a new PR to fix this bug, https://github.com/intel/neural-compressor/pull/1000. So I am going to close this one. Thanks for your submission!

**Pull request closed without merging**